syntax = "proto3";

package eleitor;

message Etapa {
    int32 etapa = 1;
    string turno = 2;
}

message Curso {
    int32 id = 1;
    string nome = 2;
    string descricao = 3;
    Etapa etapa = 4;
}

message Vinculo {
    int32 id = 1;
    string matricula = 2;
    TipoVinculo tipo = 3;
    int32 id_pessoa = 4;
    Curso curso = 5;
}

enum TipoVinculo {
    TIPO_DESCONHECIDO = 0;
    DISCENTE = 1;
    DOCENTE = 2;
}

message Pessoa {
    int32 id = 1;
    string cpf = 2;
    string email = 3;
    string data_nascimento = 4;
    string nome = 5;
    repeated Vinculo vinculos = 6;
}

message Elegivel {
    Pessoa pessoa = 1;
    int32 foto_id = 2;
}

service ElegivelService {
  rpc EnviarElegivel (Elegivel) returns (EnviarElegivelResponse);
}

message EnviarElegivelResponse {
  string mensagem = 1;
}
